src/main/webapp/script.js

```diff
@@ -533,39 +537,96 @@ function hideDiffs(cy, elems, deletedNodes, deletedEdges, addedNodes, addedEdges
 }
 
 /**
- * Zooms in on specific node
+ * Searches a cytoscape graph (cy) based on type (node or token)
+ * using a specified search function.
+ * Returns the result of the search.
+ */
+function searchAndHighlight(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText = "";
+  const query = document.getElementById(type + '-search').value;
+  let result;
+  if (query !== "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      highlightElements(cy, result);
+    } else {
+      errorText = type + " does not exist.";
+    }
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
+}
+
+/**
+ * Finds specific node from query and zooms in on it.
+ * Returns node if it exists
  */
 function searchNode(cy, query) {
-  // reset nodes to default color
+  let target = cy.$id(query);
+  if (target.length != 0) {
+    cy.fit(target, 50);
+    return target;
+  }
+}
+/**
+ * Constructs list of nodes that contain specified token
+ * and zooms in.
+ * Returns list of nodes that contain token
+ */
+function searchToken(cy, query) {
+  let target = cy.collection();
   cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
+    if (node.data().tokens.includes(query)) {
+      target = target.add(node);
+    }
   });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
-    cy.fit(target, 50);
-    return true;
-  } else {
-    // fits all nodes on screen
-    cy.fit(cy.nodes(), 50);
-    return false;
+  if (target.length > 0) {
+    const showNode = target[0][0];
+    showNode.tip.show();
+    return target;
   }
 }
 
 /**
- * Finds element in cy graph by id
+ * Highlights collection of nodes and edges
  */
-function findNodeInGraph(cy, id) {
-  if (id.length != 0) {
-    const target = cy.$('#' + id);
-    if (target.length != 0) {
-      return target;
+function highlightElements(cy, target) {
+  cy.nodes().forEach(node => node.toggleClass('non-highlighted', true));
+
+  // highlight desired nodes
+  target.forEach(node => {
+    node.style('border-width', '4px');
+    node.toggleClass('non-highlighted', false);
```

Comment:
I'm kind of confused. Shouldn't the class name of highlighted nodes be 'highlighted' as opposed to 'non-highlighted'?